### PR TITLE
Support Ruby 3.0 and Sidekiq 6.2+

### LINKED ISSUE #101: Ruby 3.0 compatibility
Why was compatibility with Ruby 3.0 dropped in sidekiq-throttled 0.14.1 (74dbf05a5a962531c14e107a61d943daf337c8e2)? Are there plans to add it back?

### PR DESCRIPTION
**Pull request by ybiquitous**
Close #101

You can see the CI status also on ybiquitous/sidekiq-throttled#1 (my fork).

### CODE REVIEW

**Comment by bvogel** [comment]
I have no real deep insights, but reviewing this it LGTM. Nice work @ybiquitous

**Review by zrod — COMMENTED** [review]
Thanks for the PR @ybiquitous !

I have no ties with this gem but decided to fork it myself, add ruby 3 support and play around a bit, mostly to learn more about it and do a sanity check to be sure that most things behave well.

I believe you [mentioned](https://github.com/sensortower/sidekiq-throttled/issues/101#issuecomment-941798579) some tests were failing, not sure if helpful but I found a few things while testing locally:

Some specs in this file were failing:
https://github.com/ybiquitous/sidekiq-throttled/blob/ruby-3/spec/sidekiq/throttled/strategy_collection_spec.rb#L4

I updated it to
```ruby
let(:collection) { described_class.new(strategies, **options) }
```

I had issues with Capybara and was only able to get all tests to run and pass after installing Chrome - probably something funky with my machine though.

Cheers!

**Comment by ybiquitous** [comment]
@zrod Thank you for the advice! I've pushed the commits 2939428 and b431dc3 following the advice!

File: spec/sidekiq_helper.rb [hunk 1]
```diff
@@ -17,6 +18,8 @@ def jid
 Sidekiq.configure_server(&configure_redis)
 Sidekiq.configure_client(&configure_redis)
 
+Sidekiq::Web.use Rack::Session::Cookie, :secret => SecureRandom.hex(32), :same_site => true, :max_age => 86_400
```
Comment (zrod) [inline spec/sidekiq_helper.rb]: Hey @ybiquitous 
I checked the CI output and noticed that it was pulling Sidekiq 6.3.1 which is odd as this gem is set up to support up to 6.1:

```
/home/runner/work/sidekiq-throttled/sidekiq-throttled/vendor/bundle/ruby/2.6.0/gems/sidekiq-6.3.1/lib/sidekiq/web.rb:154: warning: global variable `$TESTING' not initialized
```

https://github.com/ybiquitous/sidekiq-throttled/runs/4424654694?check_suite_focus=true

The tests pass locally so I went dig into the CI file and noticed that it has an outdated env name for the  `ruby/setup-ruby` action (`GEMFILE`), I believe if you rename the reference below to `BUNDLE_GEMFILE`, CI tests will pass:
https://github.com/sensortower/sidekiq-throttled/blob/7f562c34fdaa7ec1d2c0e5ebb7c2e5c4bf8573a8/.github/workflows/ci.yml#L28

Then I believe it's safe to remove this line (21) + line 6.

(I couldn't find exactly when the change was made but you can find more details in that [GH action repo](https://github.com/ruby/setup-ruby#matrix-of-gemfiles))

**Comment by ybiquitous** [comment]
My fork PR's CI status is all green! 🎉 

https://github.com/ybiquitous/sidekiq-throttled/pull/1/checks

File: spec/sidekiq_helper.rb (hunk 1 above)
Comment (ybiquitous) [reply spec/sidekiq_helper.rb]: @zrod Thank you and sorry for the late response!

I've added the Sidekiq version check like this:

```ruby
if Gem::Version.new(Sidekiq::VERSION) >= Gem::Version.new("6.2.0")
  ...
end
```

**Comment by ybiquitous** [comment]
@ixti Could you review this PR, please?

**Comment by zrod** [comment]
> My fork PR's CI status is all green! 🎉
> 
> https://github.com/ybiquitous/sidekiq-throttled/pull/1/checks

Good stuff, I see that you also added Sidekiq 6.3 which is great! I think the only thing missing would be updating the Readme but I will let the official maintainers take it from here.

Thank you @ybiquitous !

**Comment by ixti** [comment]
@ybiquitous Heya, sorry missed this one. Will review today/tomorrow.

**Review by ixti — APPROVED** [review]
HUGE! Thank you very much! Looks awesome!

**Comment by ixti** [comment]
@ybiquitous thank you for your contribution and patience.

**Comment by ybiquitous** [comment]
@ixti Never mind. Thank you for the review and merge! 😄

**Comment by bvogel** [comment]
@ixti thank you so much for reviewing and merging! Mind doing a release as well?

**Pull request merged**